### PR TITLE
Make assorted improvements to automations and workflows

### LINKED ISSUE #1090: "Add stack label" CI/CD step fails on PRs from forks
## Description

The auth token we provide for CI/CD workflow does not have the permissions necessary to run on forks.
This means that that the CI/CD on forks is failing.

## Reproduction

1. See https://github.com/WordPress/openverse/actions/runs/4562891150/jobs/8050684128?pr=1062

## Environment

https://github.com/WordPress/openverse/blob/839418fabb0972ec3c3ccf48c72fd64cbba03cb9/.github/workflows/ci_cd.yml#L148

Priority high because the community PRs are failing CI.

### PR DESCRIPTION
**Pull request by dhruvkb**
## Fixes

Fixes #1090 by @obulat

## Description

This PR is a collection of many different improvements to our workflows and automations.

- Removes `actionlint` workflow and associated code as that is a part of `pre-commit`.
- Removes Python code that was unused since we upgraded to Projects v2 and disabled old automations.
- Migrates PR labelling to Node-based approach that utilises existing PR project automation architecture
	- Removes Python code made redundant by this change.
- Moves utils into `utils/` and one-time script to `utilities`.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.
- [ ] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: automations/js/src/label_pr.mjs [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+
+  // Finally we commit all label IDs to the PR via a mutation. GitHub will only
+  // add the new IDs we provide, no existing label will be changed.
+  await pr.addLabels(Array.from(finalLabels.ids))
+}
+
+import { Octokit } from '@octokit/rest'
+let octokit = new Octokit({ auth: process.env.ACCESS_TOKEN })
+let core = { info: console.log }
+await main(octokit, core)
```
Comment (sarayourfriend) [inline automations/js/src/label_pr.mjs:122]: Is this leftover testing code? With this in, won't `main` run twice, once when the module is imported and then again when the workflow calls `main`? Or does `import` work differently and prevent that?

**Review by sarayourfriend — APPROVED** [review]
LGTM! One question for what might be leftover testing code but otherwise a nice cleanup!

Once we do this, we could clean up the three removed workflow runs from the repository so that they no longer show in the actions workflow list.

File: automations/js/src/label_pr.mjs (hunk 1 above)
Comment (dhruvkb) [reply automations/js/src/label_pr.mjs]: Deleted in ee7ffd9.

File: automations/js/src/utils/id_set.mjs [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+ */
+export class IdSet {
+  constructor() {
+    this.items = []
+    this.ids = new Set()
+  }
+
+  /**
+   * Add an item to the set. The item must have an `id` field.
+   * @param item {object | {id: string}} the item to add
```
Comment (obulat) [inline automations/js/src/utils/id_set.mjs:14]: TIL that object can be typed like this in JSDoc.

File: .github/workflows/pr_automations.yml [hunk 3]
```diff
@@ -65,10 +65,18 @@ jobs:
...
+            await main(github)
+
       - name: Perform PR automations
         uses: actions/github-script@v7
         with:
           github-token: ${{ secrets.ACCESS_TOKEN }}
           script: |
             const { main } = await import('${{ github.workspace }}/automations/js/src/project_automation/prs.mjs')
-            await main(github)
+            await main(github, core)
```
[the remark this answers is not in the document]
Comment (dhruvkb) [reply .github/workflows/pr_automations.yml:82]: `core` is a global variable in the `actions/github-script` space. What this action seems to run these lines inside a space where a bunch of useful variables are already in-scope.

**Pull request merged**